src/app/main-column/main-column.component.ts

```diff
@@ -79,7 +79,11 @@ export class MainColumnComponent extends GraphExplorerComponent implements After
 
     public ngAfterViewInit(): void {
         // Init telemetry
-        telemetry.initialize();
+        const { mscc }: any = window;
+
+        if (mscc && mscc.hasConsent()) {
+            telemetry.initialize();
```

Comment:
Now that we have conditional initialization, what happens when the telemetry object is not initialized and is used elsewhere in the code?

Reply:
I have updated the implementation to use a config property to turn off/on telemetry. It shouldn't matter whether the telemetry object is initialize or not.